### ISSUE TITLE
Use code mining to allow user to update the classpath via the Gradle editor

### ISSUE THREAD

**Issue by vogella** [eclipse/buildship#712]
If a user edits the build.gradle file he can update the classpath via right-mouse click. It would be nice if this can be done inline. For example, for the target editor I'm working to allow its activation via the text editor and code mining.

![activate](https://user-images.githubusercontent.com/139910/42211901-f4220ac6-7eb4-11e8-805c-f35f743fc676.png)

The related code changes can be found here: https://git.eclipse.org/r/#/c/125149/

It is relatively simple to provide this (see Gerrit) and I think Gradle users would benefit from this.

**Comment by oehme** [comment]
What does inline mean? What is "code mining" in this context? I can't really tell from the screenshot. Maybe a screencast would help?

**Comment by vogella** [comment]
Ah, sorry, I assume everyone knows about the Eclipse 4.8 code mining API.

Code mining are called code lenses in VSCode and allow to enhance the source code with additional information. In the above screenshot Set as Active Target Platform is not part of the source file, it is added as additional information and action which (should) allow to active the target platform.

Other, more complex examples are code mining for JDT, see https://github.com/angelozerr/jdt-codemining which contain a lot of animated gifs. The Gerrit from above should give you a hint how to implement this, if desired.

**Comment by oehme** [comment]
Got it, thanks! That looks useful indeed. Would be a great community contribution.

**Comment by vogella** [comment]
@oehme I could contribute the code mining without the specific Gradle logic. Would that be helpful? I guess it will be easy for you to add the buildship specific logic and it (should be) easy for me to add the code mining header annotation.

**Comment by vogella** [comment]
@donat WDYT?

**Comment by vogella** [comment]
Today Donat and I implemented this as a prototyp on EclipseCon.

Donát Csikós <notifications@github.com> schrieb am Mi., 17. Okt. 2018,
14:23:

**Comment by donat** [comment]
FFR the prototype is pushed to the [`donat/code-mining/prototype`](https://github.com/eclipse/buildship/tree/donat/code-mining/prototype) branch.

**Comment by vogella** [comment]
Any plans to release that?

Am Mo., 29. Okt. 2018, 10:41 hat Donát Csikós <notifications@github.com>
geschrieben:

**Comment by donat** [comment]
@vogella I've cleaned up the prototype code and created a new [PR](#847) from it. There are a couple of things which need to be sorted out before we can merge this:
- Does the extension look correctly on other platforms and themes?
- How can we disallow deleting the shortcut in the editor?
I would like to ask you to do a review on #847 (if you self-request a review then you can add your comments).
